packages/openneuro-server/src/routes.ts

```diff
@@ -157,6 +163,48 @@ const routes = [
     middleware: [noCache, orcid.authCallback],
     handler: jwt.authSuccessHandler,
   },
+
+  // GitHub authentication route
+  {
+    method: "get",
+    url: "/auth/github",
+    middleware: [storeRedirect],
+    handler: (req, res, next) => {
+      const redirectTo = req.query.redirectTo || "/"
+      passport.authenticate("github", {
+        state: encodeURIComponent(redirectTo),
+      })(req, res, next)
+    },
+  },
+
+  // GitHub OAuth callback route
+  {
+    method: "get",
+    url: "/auth/github/callback",
+    handler: (req, res, next) => {
+      passport.authenticate("github", (err, user) => {
+        if (err) {
+          // eslint-disable-next-line
+          console.error("GitHub Auth Error:", err)
+          return res.redirect("/login?error=github_auth_failed")
+        }
+        if (!user) return res.redirect("/login")
+
+        // Store JWT in a secure cookie
+        res.cookie("jwt", user.jwt, {
+          httpOnly: true,
+          secure: process.env.NODE_ENV === "production",
+          sameSite: "strict",
+        })
```

Comment:
This looks like a bad idea, why is there a need to set the JWT on a cookie here? It should be set already on the accessToken cookie when this callback resolves.

Reply:
ah, thanks. that has been removed.